pkg/messaging/message.go

```diff
@@ -11,3 +21,44 @@ type Message struct {
 func (m *Message) IsBroadcast() bool {
 	return m.To == ""
 }
+
+type controlPayload struct {
+	Type string `json:"type"`
+	Peer string `json:"peer"`
+}
+
+func NewPeerDisconnected(logger logging.Logger, peerUID string) Message {
+	payload := controlPayload{
+		Type: ctrlDisconnected,
+		Peer: peerUID,
+	}
+
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		logger.Error("failed to marshal control payload", logging.Fields{"type": "peer_disconnected", "peer": peerUID})
```

Comment:
"type" logging field is bit too general to its meaning here. I'd prefer message_type.